joblib/my_exceptions.py

```diff
@@ -1,125 +1,5 @@
-"""
-Exceptions
-"""
-# Author: Gael Varoquaux < gael dot varoquaux at normalesup dot org >
-# Copyright: 2010, Gael Varoquaux
-# License: BSD 3 clause
-from ._compat import PY3_OR_LATER
-
-
-class JoblibException(Exception):
-    """A simple exception with an error message that you can get to."""
-    def __init__(self, *args):
-        # We need to implement __init__ so that it is picked in the
-        # multiple heritance hierarchy in the class created in
-        # _mk_exception. Note: in Python 2, if you implement __init__
-        # in your exception class you need to set .args correctly,
-        # otherwise you can dump an exception instance with pickle but
-        # not load it (at load time an empty .args will be passed to
-        # the constructor). Also we want to be explicit and not use
-        # 'super' here. Using 'super' can cause a sibling class method
-        # to be called and we have no control the sibling class method
-        # constructor signature in the exception returned by
-        # _mk_exception.
-        Exception.__init__(self, *args)
-
-    def __repr__(self):
-        if hasattr(self, 'args') and len(self.args) > 0:
-            message = self.args[0]
-        else:
-            message = ''
-
-        name = self.__class__.__name__
-        return '%s\n%s\n%s\n%s' % (name, 75 * '_', message, 75 * '_')
-
-    __str__ = __repr__
-
-
-class TransportableException(JoblibException):
```

Comment:
maybe a way to do the deprecation would be to move this exception `WorkerInterrupt` and `TransportableExecption` in a private module `_joblib_exception` and leave them importable from here with a deprecation warning?

Reply:
Actually, to just move `WorkerInterrupt` in the new private module and deprecate the whole `my_exceptions` module

Reply:
But maybe it can be useful for third-party to be able to `except WorkerInterrupt`? I am not sure.

Reply:
I would just deprecate `TransportableException` as part of this PR.